torch/csrc/jit/codegen/cuda/scheduler/pointwise.h

```diff
@@ -3,18 +3,31 @@
 #include <ATen/core/ivalue.h>
 
 #include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/pointwise_heuristic.h>
 
 namespace torch {
 namespace jit {
 namespace fuser {
 namespace cuda {
 
-// return true or false on whether given fusion could be scheduled;
-TORCH_CUDA_CU_API bool scheduleFusion(
+class ExpressionEvaluator;
+
+TORCH_CUDA_CU_API c10::optional<PointwiseParams> getPointwiseHeuristics(
+    Fusion* fusion,
+    const at::ArrayRef<c10::IValue>& runtime_inputs);
+
+TORCH_CUDA_CU_API c10::optional<PointwiseParams> getPointwiseHeuristics(
```

Comment:
Is the one without the ExpressionEvaluator argument just to save overhead for validating existing kernel for dynamic shapes?

Reply:
Yes, expression evaluator is so we don't generate a new one when we analyze outputs to check if we can vectorize them.